src/ArbiterClientShim.h

```diff
@@ -0,0 +1,90 @@
+/* Copyright (c) 2015-2017 Stanford University
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR(S) DISCLAIM ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL AUTHORS BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#ifndef ARBITER_CLIENT_SHIM_H
+#define ARBITER_CLIENT_SHIM_H
+
+#include <stdint.h>
+#include <vector>
+
+#include "Semaphore.h"
+#include "CoreArbiterClient.h"
+
+
+namespace Arachne {
+
+typedef int core_t;
+
+/**
+  * This class functions as a shim for the CoreArbiter client so that Arachne
```

Comment:
It isn't clear what you mean by "shim". In addition, comments should use different words from the name of the thing being documented. Is this an alternative to CoreArbiterClient that eliminates the use of the core arbiter? I'm not sure "Shim" is the right term to use in the class name either. Let's discuss.